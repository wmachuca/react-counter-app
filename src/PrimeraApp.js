import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({ saludo, subtitulo }) => {
    return (
        <Fragment>
            <h1>{saludo}</h1>
            <p>{subtitulo}</p>
        </Fragment>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
    subtitulo: PropTypes.string
}

PrimeraApp.defaultProps = {
    saludo: 'Hola Mundo',
    subtitulo: 'Mi primera aplicaci√≥n'
}

export default PrimeraApp;

/*
    Los componentes en React, se pueden manejar de dos maneras, inicialmente se manajeban con
    clases pero ahora se usa con componentes funcionales, estos componentes funcionales antes
    son los que se llamaban stateless funcional components, pero como ahora por medio de los
    hooks si se pueden manejar estados en los componentes, por eso solo se llaman funcional
    components y es la manera que React recomienda el uso de componentes.
*/