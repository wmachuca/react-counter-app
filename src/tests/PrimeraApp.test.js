// import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';
import { shallow } from 'enzyme';

describe('Pruebas de PrimeraApp.js', () => {

    // test('Debe mostrar el mensaje `Hola, soy Wilmer` ', () => {
    //     const saludo = 'Hola, soy Wilmer';
    //     const { getByText } = render(<PrimeraApp saludo={saludo} />);
    //     expect(getByText(saludo)).toBeInTheDocument();
    // })

    test('Debe mostrar <PrimeraApp /> correctamente', () => {
        const saludo = 'Hola, soy Wilmer';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);
        expect(wrapper).toMatchSnapshot();
    })
    
    test('Debe mostrar el subtitulo enviado por props', () => {
        const saludo = 'Hola, soy Wilmer';
        const subtitulo = 'Soy un subtitulo';
        const wrapper = shallow(
            <PrimeraApp 
                saludo={saludo}
                subtitulo={subtitulo} 
            />
        );
        const textoParrafo = wrapper.find('p').text();
        expect(textoParrafo).toBe(subtitulo);
    })


})
