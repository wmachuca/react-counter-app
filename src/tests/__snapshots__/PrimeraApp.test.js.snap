// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de PrimeraApp.js Debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Wilmer
  </h1>
  <p>
    Mi primera aplicaci√≥n
  </p>
</Fragment>
`;
